### ISSUE TITLE
Dashes in column names seem to not sync properly

### ISSUE THREAD

**Issue by iand675** [org-trello/org-trello#289]
Hi, when I try to run org-trello-sync-buffer, I receive an error:
```
ou can run the command `org-trello-sync-buffer' with C-c o s
org-trello - List of errors:
 - Setup problem.
Either you did not connect your org-mode buffer with a trello board, to correct this:
  * attach to a board through C-c o I or M-x org-trello-install-board-metadata
  * or create a board from scratch with C-c o b or M-x org-trello-create-board-and-install-metadata).
Either your org-mode's todo keyword list and your trello board lists are not named the same way (which they must).
For this, connect to trello and rename your board's list according to your org-mode's todo list.
Also, you can specify on your org-mode buffer the todo list you want to work with, for example: #+TODO: TODO DOING | DONE FAIL (hit C-c C-c to refresh the setup)
```

When I look at the properties, it looks like actual dashes in the column names aren't being escaped, so they're treated as spaces. For example the board named `Shipped Aug 26 2015 - Sep 1 2015` in Trello is edited to `Shipped-Aug-26-2015---Sep-1-2015`. Note that the `-` in between Aug 26 2015 and Sep 1 2015 is not escaped in any way.

Here's the anonymised info:

```
:PROPERTIES:
#+PROPERTY: board-name App Active Dev
#+PROPERTY: board-id scramble
#+PROPERTY: Shipped-Aug-26-2015---Sep-1-2015 scramble
#+PROPERTY: Production-Testing scramble
#+PROPERTY: Code-Review scramble
#+PROPERTY: Active-Aug-26---Sep-1-2015 scramble
#+PROPERTY: Backlog-(Next-Up) scramble
#+PROPERTY: Needs-Input/Decisions scramble
#+PROPERTY: Grab-Bag-(Tech-Debt-+-Broken-Windows) scramble
#+TODO: Grab-Bag-(Tech-Debt-+-Broken-Windows) Needs-Input/Decisions Backlog-(Next-Up) Active-Aug-26---Sep-1-2015 Code-Review Production-Testing Shipped-Aug-26-2015---Sep-1-2015
#+PROPERTY: orgtrello-user-wibble 12354359365
#+PROPERTY: :green XS
#+PROPERTY: :yellow Small
#+PROPERTY: :orange Medium
#+PROPERTY: :red Large
#+PROPERTY: :purple X-Large
#+PROPERTY: :blue Changelog Minor
#+PROPERTY: orgtrello-user-me wibble
:END:
```

**Comment by ardumont** [comment]
Hello,

It'd be great if you could <kbd>M-x org-trello-bug-report<kbd> and paste the result here, thanks.
(http://org-trello.github.io/contributions.html#automatic-recommended)

For the current problem, can you please give your actual list names?

Thanks in advance.

Cheers,

**Comment by ardumont** [comment]
Mmm, i'm not sure this is the problem...
I wonder if the properties have been reloaded after writing inside the buffer at the end of routine to install the board and list ids.

You can either try:
- keep your buffer open and <kbd>M-: (orgtrello-action-reload-setup) RET<kbd>
(Note: M stands for meta)

- or close the current buffer (or revert it), activate org-trello again

Then check `org-trello-sync-buffer` again.

Cheers,

**Comment by iand675** [comment]
Hi, those are the actual list names above.

System information:
- system-type: gnu/linux
- locale-coding-system: utf-8-unix
- emacs-version: GNU Emacs 24.5.2 (x86_64-unknown-linux-gnu)
 of 2015-08-14 on localhost
- org version: 8.2.10
- org-trello version: 0.7.4
- org-trello path: /home/ian/.emacs.d/elpa/org-trello-20150824.1549/org-trello.el

I installed from melpa.

Both options presented above didn't fix the issue. I've verified that other simpler boards sync fine, but not this one.

**Comment by iand675** [comment]
These are the lists with ids too, if that's what you were asking for:

```
#+PROPERTY: board-name App Active Dev
#+PROPERTY: board-id 556e1ec78210c8e61e2016dd
#+PROPERTY: Shipped-Aug-26-2015---Sep-1-2015 55ddb9cd9aee09343063a751
#+PROPERTY: Production-Testing 55772a98d0dbaef083814d1e
#+PROPERTY: Code-Review 55772a90565026cd148d759f
#+PROPERTY: Active-Aug-26---Sep-1-2015 55ddb9a4a62e0a0830f4d9c4
#+PROPERTY: Backlog-(Next-Up) 556e1f8e5d3dd3907baaa1f9
#+PROPERTY: Needs-Input/Decisions 55ad31123860045bbed5f41a
#+PROPERTY: Grab-Bag-(Tech-Debt-+-Broken-Windows) 559ee81b51b5588365b3368c
#+TODO: Grab-Bag-(Tech-Debt-+-Broken-Windows) Needs-Input/Decisions Backlog-(Next-Up) Active-Aug-26---Sep-1-2015 Code-Review Production-Testing Shipped-Aug-26-2015---Sep-1-2015
```

**Comment by ardumont** [comment]
Hello again,

# TL;DR;

Bug on org-trello's side with `(` and `)` in list name...

Here is a temporary solution. Try adding this in your scratch buffer:
```lisp
(defun orgtrello-controller--convention-property-name (name)
  "Given a NAME, use the org property convention used in org-file headers."
  (->> name
       (replace-regexp-in-string " " "-")
       (replace-regexp-in-string "(" "")
       (replace-regexp-in-string ")" "")))
```

Then C-c C-c on it or <kbd>M-x eval-buffer RET</kbd>
This will monkey-patch your current installation.
Then try again your sync.
Is it ok?

It worked on my part.

# Investigation Detail

> These are the lists with ids too, if that's what you were asking for:

I meant for your board list's real names (on trello's side).
I made an educated guess, i supposed it was space instead of dash :D 

I created a board with the same list name (as per my hypothesis).
I hit the problem too.

By removing the ( and ) from the list with those.
`C-c o u` to refresh board metadata on buffer.
+ create a new card, a sync to trello with that card was ok.

By adding back the ( in the last one `Grab-Bag-Tech-(Debt-+-Broken-Windows)`.
C-c o u` again.
The sync to trello was still ok (strangely).

By adding back the ( in `Backlog-(Next-Up)`.
`C-c o u` again.
The sync to trello was ko this time as per the message you sent initially...

--

Since trello's api works only with list ids, the problem must lie in trello's data model.
Between the parsing of the properties and before the http request.

And i think i found the issue, the ( and ) must disappear from board list name (like spaces).

Here is the output of some internal function that help showing local buffer `state`:

<kbd>M-: (orgtrello-setup-display-current-buffer-setup) RET</kbd>

```lisp
(:user-logged-in "ardumont"
 :org-keyword-trello-list-names ("Shipped-Aug-26-2015"
 "Production-testing"
 "Code-Review"
 "Active-Aug-26---Sep-1-2015"
 "Backlog-"                                                         <- interesting, this got split at the (
 "Needs-Input-Decisions"
 "Grab-Bag-Tech-Debt-+-Broken-Windows")
 
 :org-keyword-id-name #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data (
   "55de0ad90a224716ff8ce9b7" "Shipped-Aug-26-2015"
   "55de0b3dcb39bfbdd1f80cbd" "Production-testing"
   "55de0b43117d6fbc014e3e26" "Code-Review"
   "55de0b51b001798f7bf7fa0a" "Active-Aug-26---Sep-1-2015"
   nil "Grab-Bag-Tech-Debt-+-Broken-Windows"                        <- here again, a key is not supposed to be nil
   "55de0b66c7dd8bc82ff92d12" "Needs-Input-Decisions" ...)))
```

I already clean up the space for the org properties to work #+property: <blablawithoutspace-is-a-key> <value>.
But this seems not sufficient now. The `(` and `)` seem to fuck things up :D.

Cheers,

**Comment by iand675** [comment]
Hi @ardumont thanks for the *really thorough* bug report! The sync command doesn't show an error now. However, it's still not syncing successfully:

```
Loading /home/ian/.emacs.d/.trello/ianduncan2.el (source)...done
org-trello - Synchronizing org-mode file to board 'App Active Dev'. This may take some time, some coffee may be a good idea...
org-trello - No card(s) to sync.
```

I'm not sure whether that's a regression introduced by your suggestion or a separate issue.

**Comment by ardumont** [comment]
> I'm not sure whether that's a regression introduced by your suggestion or a separate issue.

I'm not sure either.

What does `C-c o d` gives you?

Cheers,

**Comment by iand675** [comment]
org-trello - Account 'ianduncan2' configured! Everything is ok!

**Comment by ardumont** [comment]
Stating the obvious to have it out of the way, you do have stuff on your trello board that you want to dump into your org buffer, is that it?

**Comment by iand675** [comment]
Yes, that's what I have in mind. I'm trying to get my work team's trello board into this buffer, and there are currently 46 cards in total.

**Comment by ardumont** [comment]
Hello,

> Yes, that's what I have in mind. I'm trying to get my work team's trello board into this buffer, and there are currently 46 cards in total.

Right. Aligned in thought :D 

Can you please check the following?

<kbd>M-x ielm</kbd> (emacs' repl).

And then insert the following:
```lisp
(let ((org-trello-file "/home/tony/repo/perso/org-trello/TESTS-simple.org")
      (orgtrello-log-level orgtrello-log-trace))
  (with-temp-buffer
    (insert-file-contents org-trello-file)
    (org-mode)
    (org-trello-mode)
    (orgtrello-controller-load-keys)
    (-> (orgtrello-buffer-board-id)
        orgtrello-api-get-full-cards
        (orgtrello-query-http-trello 'sync))))
```
Note: Replace the string value "/home/tony.../" for org-trello-file by the file we are trying to debug.

Thanks for your patience.

Cheers,
